### ISSUE TITLE
Rule schema default not working with nameless option

### ISSUE THREAD

**Issue by GMartigny** [eslint/eslint#11749]
**Tell us about your environment**

* **ESLint Version:**5.16.0
* **Node Version:**11.13.0
* **npm Version:**6.7.0

**What parser (default, Babel-ESLint, etc.) are you using?**
Babel-ESLint

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

While creating a rule, I wanted to use the `meta.schema` default option. But it doesn't seems to work with unnamed options.

```js
module.exports = {
	context => {
	    console.log(context.options); // => []
    },
	meta: {
		schema: [{
			type: 'integer',
			default: 5
		}]
	}
};
```

```bash
# config => "rule": [1]
eslint 
```

**What did you expect to happen?**
As I specified a `default`, I expected it to set the value of the options when `undefined`, that `console.log` display `[ 5 ]`. 

**What actually happened? Please include the actual, raw output from ESLint.**
Whatever the `default` value or `type`, it never shows in the `console.log`. However, when I use named options, it works as expected.

```js
module.exports = {
	context => {
	    console.log(context.options); // => [ { message: 5 } ]
    },
	meta: {
		schema: [{
			type: 'object',
			properties: {
				whatever: {
					type: 'integer',
					default: 5
				}
			}
		}]
	}
};
```
```bash
# config => "rule": [1, {}]
eslint 
```

**Are you willing to submit a pull request to fix this bug?**
Why not, but I have no knowledge of the codebase.

**Comment by kaicataldo** [comment]
We use [ajv](https://github.com/epoberezkin/ajv) for JSON schema validation, which does have the [option for setting default values](https://github.com/epoberezkin/ajv#assigning-defaults). It looks like we're running ajv with [this option set to `true`](https://github.com/eslint/eslint/blob/219aecb78bc646d44bad27dc775a9b3d3dc58232/lib/shared/ajv.js#L21), but the option itself is limited to being used only in `properties` and `items`.

From the link above:

> With option useDefaults Ajv will assign values from default keyword in the schemas of properties and items (when it is the array of schemas) to the missing properties and items

The example you shared above seems to behave consistently with this, so I believe it's working as intended.
